### PR TITLE
Cleanup using C++17 features

### PR DESCRIPTION
**Pull request by codereport**
Small cleanup PR

### CODE REVIEW

File: cpp/include/cudf_test/timestamp_utilities.cuh [hunk 1]
```diff
@@ -55,7 +55,6 @@ inline cudf::test::fixed_width_column_wrapper<T, int64_t> generate_timestamps(in
   auto lhs = start.time_since_epoch().count();
   auto rhs = stop.time_since_epoch().count();
 
-  // When C++17, auto [min, max] = std::minmax(lhs, rhs)
   auto min   = std::min(lhs, rhs);
   auto max   = std::max(lhs, rhs);
```
Comment (ttnghia) [inline cpp/include/cudf_test/timestamp_utilities.cuh:59]: `When C++17, auto [min, max] = std::minmax(lhs, rhs)` => so why not `auto [min, max] = std::minmax(lhs, rhs)` now?

File: cpp/include/cudf_test/timestamp_utilities.cuh [hunk 2]
```diff
@@ -55,7 +55,6 @@ inline cudf::test::fixed_width_column_wrapper<T, int64_t> generate_timestamps(in
...
   auto range = max - min;
```
Comment (ttnghia) [inline cpp/include/cudf_test/timestamp_utilities.cuh:60]: Suggest adding `const` qualifier for the variables.

File: cpp/src/merge/merge.cu [hunk 3]
```diff
@@ -297,7 +297,6 @@ struct column_merger {
                       row_order_.end(),
                       merged_view.begin<Element>(),
                       [d_lcol, d_rcol] __device__(index_type const& index_pair) {
-                        // When C++17, use structure bindings
                         auto side  = thrust::get<0>(index_pair);
                         auto index = thrust::get<1>(index_pair);
```
Comment (ttnghia) [inline cpp/src/merge/merge.cu:301]: So why not structure binding now?

File: cpp/tests/wrappers/timestamps_test.cu [hunk 4]
```diff
@@ -83,10 +83,7 @@ TYPED_TEST(ChronoColumnTest, ChronoDurationsMatchPrimitiveRepresentation)
...
   // round-trip through the host to copy `chrono_col` values
   // to a new fixed_width_column_wrapper `primitive_col`
-  // When C++17, use structured bindings
-  thrust::host_vector<Rep> chrono_col_data;
-  std::vector<cudf::bitmask_type> chrono_col_mask;
-  std::tie(chrono_col_data, chrono_col_mask) = to_host<Rep>(chrono_col);
+  auto [chrono_col_data, chrono_col_mask] = to_host<Rep>(chrono_col);
 
   auto primitive_col =
     fixed_width_column_wrapper<Rep>(chrono_col_data.begin(), chrono_col_data.end());
```
Comment (ttnghia) [inline cpp/tests/wrappers/timestamps_test.cu:89]: `auto const` would be better if possible.

File: cpp/include/cudf_test/timestamp_utilities.cuh (hunk 1 above)
Comment (codereport) [reply cpp/include/cudf_test/timestamp_utilities.cuh:59]: You get: `error: structured binding cannot be captured` which from googling seems to be a compiler bug?

File: cpp/tests/wrappers/timestamps_test.cu (hunk 4 above)
Comment (codereport) [reply cpp/tests/wrappers/timestamps_test.cu:89]: Done.

File: cpp/include/cudf_test/timestamp_utilities.cuh (hunk 2 above)
Comment (codereport) [reply cpp/include/cudf_test/timestamp_utilities.cuh]: Done.

File: cpp/src/merge/merge.cu (hunk 3 above)
Comment (codereport) [reply cpp/src/merge/merge.cu:301]: It seems `thrust::tuple` does not work properly with structured bindings.

File: cpp/include/cudf_test/timestamp_utilities.cuh (hunk 1 above)
Comment (ttnghia) [reply cpp/include/cudf_test/timestamp_utilities.cuh]: Oh no...

File: cpp/src/merge/merge.cu (hunk 3 above)
Comment (ttnghia) [reply cpp/src/merge/merge.cu:301]: Oh no...

**Review by ttnghia — APPROVED** [review]

**Review by mythrocks — APPROVED** [review]

**Comment by codereport** [comment]
@gpucibot merge

File: cpp/include/cudf_test/timestamp_utilities.cuh [hunk 5]
```diff
@@ -55,11 +55,10 @@ inline cudf::test::fixed_width_column_wrapper<T, int64_t> generate_timestamps(in
   auto lhs = start.time_since_epoch().count();
   auto rhs = stop.time_since_epoch().count();
 
-  // When C++17, auto [min, max] = std::minmax(lhs, rhs)
-  auto min   = std::min(lhs, rhs);
-  auto max   = std::max(lhs, rhs);
-  auto range = max - min;
-  auto iter  = cudf::detail::make_counting_transform_iterator(0, [=](auto i) {
+  auto const min   = std::min(lhs, rhs);
+  auto const max   = std::max(lhs, rhs);
```
Comment (karthikeyann) [inline cpp/include/cudf_test/timestamp_utilities.cuh:59]: Why not minmax as mentioned in the comment?

**Pull request merged**